### PR TITLE
chore: Reduce file name length to avoid risk of too long filename on checkout

### PR DESCRIPTION
**Pull request by dr1rrb**
## Other
Reduce file name length to avoid risk of too long filename on checkout

## What is the current behavior?
Cannot checkout on windows unless in a very short path or having in `.gitconfig` :

```
[core]
    longpaths = true
```

## What is the new behavior?
Reduce the risk of issue

## PR Checklist
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators.netcore.Tests/XamlCodeGeneratorTests/Given_MvvmGeneratedMembers.cs [hunk 1]
```diff
@@ -120,7 +119,7 @@ public async Task When_ObservableProperty_AttributeDoesNotExists()
 			</Page>
 			""");
 
-		var test = new Test(xamlFile)
+		var test = new MvvmTest(xamlFile, "WOPADNE")
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators.netcore.Tests/XamlCodeGeneratorTests/Given_MvvmGeneratedMembers.cs:122]: Not a fan tbh :'(

**Review by Youssef1313 — APPROVED** [review]
Approving, though the unreadable short names are unfortunate.

**Review by Youssef1313 — COMMENTED** [review]
@dr1rrb Can we instead go from full hash to only the first 5 characters or so?

That would be done here:

https://github.com/unoplatform/uno/blob/08e81f34ec1cc503d46320fb6c6b5fd13ae9444e/src/SourceGenerators/SourceGeneratorHelpers/Helpers/HashBuilder.cs#L28-L47

I'd prefer that more than the current approach.

- Hash 32 chars -> 5 chars: Saves 27 characters
- `TestOutput` -> `Out`: 7 characters save

**Comment by dr1rrb** [comment]
> Approving, though the unreadable short names are unfortunate.

Agree, it's the best approach I found to quickly and significantly reduce the file path.

> @dr1rrb Can we instead go from full hash to only the first 5 characters or so?
> 

We could, but it would require a bigger refactor and tests, while this PR is just files renaming in a test project for a quick patch.

File: global.json [hunk 2]
```diff
@@ -1,12 +1,4 @@
 {
-  "sdk": {
-    "version": "7.0.102",
-    "allowPrerelease": true,
-    "rollForward": "latestPatch"
-  },
-  "tools": {
-    "dotnet": "7.0.102"
-  },
```
Comment (Youssef1313) [inline global.json:9]: These shouldn't be removed.

Comment (dr1rrb) [reply global.json]: Oups, fixed (even if fixing the specific version is incredibly annoying, has to change it on each checkout :/)

**Comment by Youssef1313** [comment]
> We could, but it would require a bigger refactor and tests, while this PR is just files renaming in a test project for a quick patch.

I thought it should be a smaller change?

**Comment by dr1rrb** [comment]
> > We could, but it would require a bigger refactor and tests, while this PR is just files renaming in a test project for a quick patch.
> 
> I thought it should be a smaller change?

Smaller change, but bigger impact

**Pull request merged**